File: Sorting and Searching/Josephus Problem/josephusProblem.java
import java.util.Scanner;

public class josephusProblem {
    private static void solve(){
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[] arr = new int[n];
        Boolean[] eliminate = new Boolean[n]; 
        
        for(int i=1;i<=n;i++) {
            arr[i-1] = i;
            eliminate[i-1] = false;
        }

        int idx = 0;
        for(int i=0;i<n;i++){
            int cnt = 0;
            while(cnt < 2){
                if(eliminate[idx] == false){
                    cnt++;
                }
                if(cnt < 2){
                    idx = (idx+1)%n;
                }
            }
            eliminate[idx] = true;
            System.out.print(arr[idx] + " ");
        }
    }
    public static void main(String args[]){
        solve();
    }    
}

File: Sorting and Searching/Josephus Problem/josephus_problem.cpp
#include<bits/stdc++.h>
using namespace std;
#define ll long long int

void solve(){
    int n, cnt = 0;
    
    cin>>n;
    vector<int> arr, eliminated(n, 0);
    for(int i=1;i<=n;i++){
        arr.push_back(i);
    }

	int current = 0;
    for(int i=0;i<n;i++){
    	int count = 0;
    	while(count < 2){
    		if(eliminated[current] == 0){
    			count++;
    		}
    		if(count < 2){
    			current = (current + 1)%n;
    		}
    		
    	}
    	eliminated[current] = 1;
    	cout<<arr[current]<<" ";
    }
    return;
}

int main(){
	ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    long long testCases;
	testCases = 1;
	// cin>>testCases;
	while(testCases--){
		solve();
	}
	
	return 0;
}

